app/jinja_filters.py

```diff
@@ -1,41 +1,48 @@
 # coding: utf-8
 import re
+from decimal import Decimal
+from typing import Any, Mapping, Optional, Union
 
 import flask
 import flask_babel
 from babel import numbers, units
 from flask import current_app
 from jinja2 import pass_eval_context
 from markupsafe import Markup, escape
+from wtforms import SelectFieldBase
 
 from app.questionnaire.rules.utils import parse_datetime
 from app.settings import MAX_NUMBER
 
 blueprint = flask.Blueprint("filters", __name__)
 
 
-def mark_safe(context, value):
+def mark_safe(context: Any, value: str) -> str:
     return Markup(value) if context.autoescape else value
 
 
-def strip_tags(value):
+def strip_tags(value: str) -> str:
     return escape(Markup(value).striptags())
 
 
 @blueprint.app_template_filter()
-def format_number(value):
+def format_number(value: int) -> str:
+    formatted_number: str
     if value or value == 0:
-        return numbers.format_decimal(value, locale=flask_babel.get_locale())
+        formatted_number = numbers.format_decimal(
+            value, locale=flask_babel.get_locale()
+        )
+        return formatted_number
 
     return ""
 
 
-def get_formatted_address(address_fields):
+def get_formatted_address(address_fields: dict) -> str:
     address_fields.pop("uprn", None)
     return "<br>".join(address_field for address_field in address_fields.values())
 
 
-def get_formatted_currency(value, currency="GBP") -> str:
+def get_formatted_currency(value: Union[float, Decimal], currency: str = "GBP") -> Any:
```

Comment:
The return here should be str, might need explicitly set the var as a str for this to work though.